src/main/java/com/streisky/virtualshowcasebackend/controller/product/ProductController.java

```diff
@@ -1,42 +1,91 @@
 package com.streisky.virtualshowcasebackend.controller.product;
 
+import java.net.URI;
+
 import com.streisky.virtualshowcasebackend.dto.product.ProductDTO;
+import com.streisky.virtualshowcasebackend.exception.VirtualShowcaseNotFoundException;
 import com.streisky.virtualshowcasebackend.service.product.ProductService;
 import jakarta.transaction.Transactional;
 import jakarta.validation.Valid;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.data.web.PageableDefault;
-import org.springframework.web.bind.annotation.*;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
 
 @RestController
-@RequestMapping("product")
+@RequestMapping("products")
 public class ProductController {
 
     @Autowired
     private ProductService productService;
 
     @PostMapping
     @Transactional
-    public void register(@RequestBody @Valid ProductDTO productDTO) {
-        productService.save(productDTO);
+    public ResponseEntity<ProductDTO> register(@RequestBody @Valid ProductDTO productDTO) {
+        try {
+            ProductDTO savedProduct = productService.save(productDTO);
+            URI location = ServletUriComponentsBuilder
+                    .fromCurrentRequest()
+                    .path("/{id}")
+                    .buildAndExpand(savedProduct.id())
+                    .toUri();
+
+            return ResponseEntity.created(location).body(savedProduct);
+        } catch (Exception e) {
+            return ResponseEntity.badRequest().build();
```

Comment:
This way, any error will be returned to the client as a Bad Request. Ideally, the error returned should represent specific exceptions thrown on the server side. For that, you can use [RestControllerAdvice](https://www.baeldung.com/exception-handling-for-rest-with-spring). You can also check an example [here](https://github.com/JoaoPucci/athena/blob/develop/src/main/java/tech/dtech/athena/config/validation/exceptions/handler/ControllerExceptionHandler.java).

Reply:
This treatment will be done in another PR